File: 0950-Reveal-Cards-In-Increasing-Order/py/solution.py
class Solution:
    def deckRevealedIncreasing(self, deck):
        deck.sort()
        ans = []
        while deck:
            if ans:
                ans.insert(0, ans.pop())
            ans.insert(0, deck.pop())
        return ans


if __name__ == '__main__':
    print(111)

File: 0950-Reveal-Cards-In-Increasing-Order/src/Solution.java
import java.util.Arrays;
import java.util.LinkedList;

/**
 * @author qybit
 * @create 2020-12-02 11:06
 */
public class Solution {

    public int[] deckRevealedIncreasing(int[] d) {
        LinkedList<Integer> ans = new LinkedList<>();
        Arrays.sort(d);
        LinkedList<Integer> q = new LinkedList<>();
        for (int x :d ) {
            q.offerLast(x);
        }
        while (!q.isEmpty()) {
            if (!ans.isEmpty()) {
                ans.offerFirst(ans.pollLast());
            }
            ans.offerFirst(q.pollLast());
        }
        int[] res = new int[d.length];
        int i  =0;
        for (Integer x : ans) {
            res[i++]  =x;
        }
        return res;
    }
}
